transport/http2_client.go

```diff
@@ -221,7 +228,11 @@ func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (_ *Strea
 	t.hEnc.WriteField(hpack.HeaderField{Name: ":scheme", Value: t.scheme})
 	t.hEnc.WriteField(hpack.HeaderField{Name: ":path", Value: callHdr.Method})
 	t.hEnc.WriteField(hpack.HeaderField{Name: ":authority", Value: callHdr.Host})
-	t.hEnc.WriteField(hpack.HeaderField{Name: "content-type", Value: "application/grpc"})
+	ct := "application/grpc"
+	if callHdr.Codec != nil {
+		ct += "+" + callHdr.Codec.String()
```

Comment:
oh, this is where it is used. hmm.

Is this use of `Content-Type` documented somewhere?